src/main/drivers/serial.c

```diff
@@ -86,6 +86,51 @@ void serialSetMode(serialPort_t *instance, portMode_e mode)
     instance->vTable->setMode(instance, mode);
 }
 
+void serialSetCtrlLineStateCb(serialPort_t *serialPort, void (*cb)(serialPort_t *context, uint16_t ctrlLineState), serialPort_t *context)
```

Comment:
This is getting a bit out of hand. Using something like posix `ioctl` (with vararg) (http://man7.org/linux/man-pages/man2/ioctl.2.html) or `setsockopt` (pasing cmd, ptr, size) vill reduce number of virtual functions.

Driver can switch on CMD, it will be compact and easier to read ...

Reply:
Agreed that we could go down the ioctl route, however that's not suitable for registering callbacks. This isn't the normal kernel/user split. These callbacks are being called in USB ISR context.

Reply:
Yes, but exporting so many virtual functions feels bad. Only registration should be done through `setsockopt`-like call, driver will invoke callback directly ...